### PR TITLE
Port replication_pad1d to ATen

### PR DESCRIPTION
**Pull request by ezyang**
Summary: port replication_pad1d

Differential Revision: D13531920

### CODE REVIEW

File: aten/src/ATen/native/cuda/ReplicationPadding.cu [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      const PackedTensorAccessor<Dtype, 3>& input,
+      PackedTensorAccessor<Dtype, 3>& output,
+      int padL, int padR) {
+    int outputPointId = threadIdx.x + blockIdx.x * blockDim.x;
+    int plane = blockIdx.y;
+    int batch = blockIdx.z;
+    if (outputPointId >= output.size(2)) {
+    return;
+    }
+    int outputPointX = outputPointId % output.size(2);
```
Comment (t-vi) [inline aten/src/ATen/native/cuda/ReplicationPadding.cu]: It's not the problem we're looking for, but Is that a thing or copypasting from 2d? (see the if ... return above).

File: aten/src/ATen/native/cuda/ReplicationPadding.cu [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+__host__ __device__ __forceinline__ int iabs(int a) {
+      return a >= 0 ? a : -a;
+}
+
+namespace {
+template <typename Dtype>
+  __global__ void replication_pad(
+      const PackedTensorAccessor<Dtype, 3>& input,
+      PackedTensorAccessor<Dtype, 3>& output,
```
Comment (t-vi) [inline aten/src/ATen/native/cuda/ReplicationPadding.cu]: I don't think we want to pass the `PackedTensorAccessor`s by reference. To me, the main thing about them (compared to `TensorAccessor` is that they carry their sizes/strides).
I think DeviceTensor worked the same way.

Comment (ezyang) [reply aten/src/ATen/native/cuda/ReplicationPadding.cu]: Oooh, that's exactly right. I feel embarrassed for missing it!

**Comment by ezyang** [comment]
CC @lhuang04 for the comments

File: aten/src/ATen/native/cuda/ReplicationPadding.cu (hunk 2 above)
Comment (t-vi) [reply aten/src/ATen/native/cuda/ReplicationPadding.cu]: You found so many bugs in my PRs, I'm glad that after so many of them, I get to return the favour. (And I only saw it after debugging in the direction of *are the offsets into the tensors invalid?* for quite a while...)

**Pull request closed without merging**